src/index.ts

```diff
@@ -84,18 +86,20 @@ export async function newEventStore(pool: Pool, emit: Emitter): Promise<EventSto
         .digest('hex');
 
       const latestSnapshot = await pool.query(aggregateQuery, [id])
-        .then((aggResult) => Option.of(aggResult.rows[0]));
+        .then((aggResult) => Option.of(aggResult.rows[0] ? aggResult.rows[0].data : null));
 
-      const aggregatedResult = await results.rows.reduce((acc, event) => {
+      const aggregatedResult = await results.rows.reduce((acc, row) => {
         return matches.reduce((matchAcc, [validate, execute]) => {
-          if (validate(event)) {
-            return execute(matchAcc, event);
+          if (validate(row.data)) {
+            return execute(matchAcc, row.data);
           }
           return matchAcc;
         }, acc);
-      }, latestSnapshot.getOrElse(accumulator));
+      }, latestSnapshot);
 
-      await pool.query(upsertAggregateCache, [id, aggregatedResult]);
+      if (aggregatedResult.nonEmpty()) {
+        await pool.query(upsertAggregateCache, [id, aggregatedResult.get()]);
+      }
```

Comment:
If we just insert `aggregatedResult`, we insert the entire option into the DB, not the value that it holds.